docs/guides/use_checkpointing.md

```diff
@@ -1,67 +1,85 @@
 * How to save and load checkpoints when you run in multi-host scenarios with
 [`flax.training.checkpoints.save_checkpoint_multiprocess`](https://flax.readthedocs.io/en/latest/api_reference/flax.training.html#flax.training.checkpoints.save_checkpoint_multiprocess).
 
+
 ## Setup
 
 Install/upgrade Flax, JAX, [Optax](https://optax.readthedocs.io/) and [TensorStore](https://google.github.io/tensorstore/). For JAX installation with GPU/TPU support, visit [this section on GitHub](https://github.com/google/jax#installation).
 
-```{code-cell} ipython3
-!pip install -U -q flax jax jaxlib optax tensorstore
+```python
+!pip install -U -q flax jax jaxlib optax tensorstore orbax
+
+# Installing Flax and Orbax from head to use latest checkpoint features.
+# This part will be removed after both make a new release.
+!pip3 install --upgrade --no-deps --force-reinstall \
+"git+https://github.com/google/flax.git@main#egg=flax"
+!pip3 install --upgrade --no-deps --force-reinstall \
+"git+https://github.com/google/orbax.git@main#egg=orbax"
 ```
 
 Note: Before running `import jax`, create eight fake devices to mimic multi-host checkpointing in this notebook. Note that the order of imports is important here. The `os.environ["XLA_FLAGS"] = '--xla_force_host_platform_device_count=8'` command works only with the CPU backend. This means it won't work with GPU/TPU acceleration on if you're running this notebook in Google Colab. If you are already running the code on multiple devices (for example, in a 4x2 TPU environment), you can skip running the next cell.
 
-```{code-cell} ipython3
+```python
 import os
 os.environ["XLA_FLAGS"] = '--xla_force_host_platform_device_count=8'
 ```
 
-```{code-cell} ipython3
+```python
 from typing import Optional, Any
 import shutil
 
 import numpy as np
 import jax
 from jax import random, numpy as jnp
-from jax.experimental import maps, PartitionSpec, pjit
-from jax.experimental.gda_serialization.serialization import GlobalAsyncCheckpointManager
 
 import flax
 from flax import linen as nn
 from flax.training import checkpoints, train_state
 from flax import struct, serialization
+import orbax.checkpoint as orbax
+
 import optax
 ```
 
+```python
+# Orbax needs to enable asyncio in colab environment.
+!pip install nest_asyncio
+import nest_asyncio
+nest_asyncio.apply()
+```
+
 ## Save checkpoints
 
 In Flax, you save and load any given JAX [pytree](https://jax.readthedocs.io/en/latest/pytrees.html) using the `flax.training.checkpoints` package. This includes not only typical Python and NumPy containers, but also customized classes extended from [`flax.struct.dataclass`](https://flax.readthedocs.io/en/latest/api_reference/flax.struct.html#flax.struct.dataclass). That means you can store almost any data generated—not only your model parameters, but any arrays/dictionaries, metadata/configs, and so on.
 
 Create a pytree with many data structures and containers, and play with it:
 
-```{code-cell} ipython3
+```python
```

Comment:
This is cool. If the `jupytext` pre-commit test passes, we can manually change/improve all our other Markdown files.